aten/src/ATen/native/ConvUtils.h

```diff
@@ -214,4 +218,179 @@ static inline bool miopen_conv_use_channels_last(const at::Tensor& input, const
   return can_use_miopen_channels_last_2d || can_use_miopen_channels_last_3d;
 }
 
+static Tensor view_weight_2d(const Tensor& weight_) {
+  Tensor weight = weight_.contiguous();
+  if (weight.dim() == 4) {
+    const int64_t s1 = weight.size(0);
+    const int64_t s2 = weight.size(1) * weight.size(2) * weight.size(3);
+    return weight.view({s1, s2});
+  } else {
+    return weight;
+  }
```

Comment:
Is the precondition for this function that weight is either 4D or 2D? Because the if statement looks fishy.

Reply:
Yeah this is super strange. I lifted it from `aten/src/ATen/native/ConvolutionMM2d.cpp` but I actually don't need it now after changing how finput computation, so I put it back.